File: ea-domain/src/main/kotlin/com/nbe3/domain/posts/exception/PostNotFoundException.java
package com.nbe3.domain.posts.exception;

import com.nbe3.common.exception.DomainException;

public class PostNotFoundException extends DomainException {

    public static final DomainException EXCEPTION = new PostNotFoundException();

    private PostNotFoundException() {
        super(PostErrorCode.POST_NOT_FOUND);
    }
}

File: ea-domain/src/main/java/com/nbe2/domain/posts/exception/PostNotFoundException.kt
package com.nbe2.domain.posts.exception

import com.nbe2.common.exception.DomainException

class PostNotFoundException private constructor(errorCode: PostErrorCode) :
        DomainException(errorCode) {
    companion object {
        val EXCEPTION: DomainException =
                PostNotFoundException(PostErrorCode.POST_NOT_FOUND)
    }
}
// object PostNotFoundException : DomainException(PostErrorCode.POST_NOT_FOUND)
